File: Cap 05/Topico 5.3/Topico 5.3.2/Pre-Teste Fluxo Falso/Interativo/atesejaINT.swift
import Foundation

var N: Int = 0
var R: Int = 0
var RESP: Character = "S"

print("LACO PRE-TESTE COM FLUXO VERDADEIRO")
print("Multiplicacao de numero por 3 por N vezes\n")
RESP = "S"
while RESP == "S" {
    print("Entre valor numerico inteiro: ", terminator: "")
    if let input = readLine(), let numericInput = Int(input) {
        N = numericInput
        R = N * 3
        print("O numero informado X 3 = \(R)\n")
    }
    print("Deseja continuar? (S/N): ", terminator: "")
    if let response = readLine()?.uppercased() {
        RESP = response.first ?? "N"
    }
    print()
}

File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.2 - Controle Condicional Falso/Interativo/atesejaINT.py
'''
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.3.2
 *
 * Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO FALSO INTERATIVO
'''

print('LACO PRE-TESTE COM FLUXO FALSO')
print('Multiplicacao de numero por 3 por N vezes')
print()
RESP = 'S'
while (not (RESP != 'S')): # simulacao "ate_seja (RESP <> "S") efetue"
  N = int(input('Entre valor numerico inteiro: '))
  R = N * 3
  print('O numero informado X 3 =', R)
  print()
  RESP = input('Deseja continuar? (S/N): ').upper()
  print()
  # simulaçao "fim_ate_seja"
